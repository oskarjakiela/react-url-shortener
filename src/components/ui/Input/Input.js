import classnames from 'classnames';
import React from 'react';

import './Input.css';


const displayName = 'Shorty-Input';

const Input = ({
  className,
  ...props
}) => (
  <input className={classnames(displayName, className)} {...props} />
);

Input.displayName = displayName;

export default Input;
