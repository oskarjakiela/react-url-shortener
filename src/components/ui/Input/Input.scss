.Shorty-Input {
  display: inline-block;
  border: 0;
  border-radius: 3px;
  padding: 0 1.125rem;
  height: 2.625rem;
  font-family: 'Roboto', sans-serif;
  font-weight: 300;
  font-size: 1rem;
  background-color: #eaeaea;
  color: #555;
  vertical-align: middle;
  outline: none;

  &::placeholder {
    color: #bbb;
  }
}

