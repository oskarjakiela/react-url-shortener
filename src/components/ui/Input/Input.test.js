import React from 'react';
import renderer from 'react-test-renderer';

import Input from './Input';


it('<Input /> renders correctly', () => {
  const tree = renderer
    .create(<Input />)
    .toJSON();

  expect(tree).toMatchSnapshot();
});

it('<Input className="foo-bar" /> renders correctly', () => {
  const tree = renderer
    .create(<Input className="foo-bar" />)
    .toJSON();

  expect(tree).toMatchSnapshot();
});
