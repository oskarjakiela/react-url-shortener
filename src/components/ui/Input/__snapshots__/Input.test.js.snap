// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Input /> renders correctly 1`] = `
<input
  className="Shorty-Input"
/>
`;

exports[`<Input className="foo-bar" /> renders correctly 1`] = `
<input
  className="Shorty-Input foo-bar"
/>
`;
